Essentials/src/main/java/com/earth2me/essentials/MetaItemStack.java

```diff
@@ -543,9 +543,9 @@ public void addPotionMeta(final CommandSource sender, final boolean allowShortNa
                 pmeta.addCustomEffect(pEffect, true);
                 stack.setItemMeta(pmeta);
                 if (VersionUtil.getServerBukkitVersion().isHigherThanOrEqualTo(VersionUtil.v1_9_R01)) {
-                    if (isSplashPotion && stack.getType() != Material.SPLASH_POTION) {
+                    if (isSplashPotion && stack.getType() != Material.SPLASH_POTION && stack.getType() != Material.TIPPED_ARROW) {
```

Comment:
This looks generally fine to me, but would prefer that you do a single check instead of two here. Can just wrap the whole if/elseif, which does add a layer of indentation but is clearer and I think that's more important.

Reply:
Alternatively we can probably avoid the entire thing by flipping the conditions here. E.g. check if isSplashPotion and the stash type is a regular potion, then convert to splash. And vice versa.

Reply:
I though about flipping the conditions first but I didn't know if this would break anything or if it was just to let the players create a splash potion using splash:true. That's a nice one yeah.

I am relatively new to this kind of stuff... I am the one who is supposed to change the code right?

Reply:
Yeah I'm 99% sure that it is just to create a splash potion with splash:true.

Essentials is just really old so back then there were not any other potion-like items. Hence the bug now, caused by the conditions being set up here in a vague way. Should definitely be more explicit.

Is it possible it could break something? Maybe, but probably not imo. Either way this code just needs to be more robust. :p